tools/shared.py

```diff
@@ -472,6 +472,15 @@ def find_temp_directory():
 def get_emscripten_version(path):
   return open(path).read().strip().replace('"', '')
 
+# Returns true if Emscripten is running in 'strict' mode, in which deprecated compiler features are not supported.
+def is_emscripten_strict():
+  if os.environ.get('EMCC_STRICT') and int(os.environ.get('EMCC_STRICT')) != 0: return True
```

Comment:
this `int` operation may throw, if the value is e.g. `a`. instead, please do `os.environ.get('EMCC_STRICT') != '0'`